.github/workflows/codeql-analysis.yml

```diff
@@ -18,6 +18,9 @@ on:
   schedule:
     - cron: '0 4 * * 6'
 
+permissions:
+  security-events: write
```

Comment:
This allows an action to update the status of a code scanning alert, this makes sense.